android/src/main/java/com/mattermost/turbolog/TurboLog.kt

```
@@ -0,0 +1,126 @@
+        rollingPolicy.context = loggerContext
+        rollingPolicy.fileNamePattern = "$logsDirectory/$logPrefix-%d{yyyy-MM-dd}.%i.log"
+        rollingPolicy.setMaxFileSize(FileSize(maximumFileSize.toLong()))
+        rollingPolicy.setTotalSizeCap(FileSize(maximumNumberOfFiles.toLong() * maximumFileSize))
+        rollingPolicy.maxHistory = maximumNumberOfFiles
+        rollingPolicy.setParent(rollingFileAppender)
+        rollingPolicy.start()
+        rollingFileAppender.setRollingPolicy(rollingPolicy)
+      } else if (maximumFileSize > 0) {
+        val rollingPolicy = FixedWindowRollingPolicy()
+        rollingPolicy.context = loggerContext
+        rollingPolicy.fileNamePattern = "$logsDirectory/$logPrefix-%i.log"
+        rollingPolicy.minIndex = 1
+        rollingPolicy.maxIndex = maximumNumberOfFiles
+        rollingPolicy.setParent(rollingFileAppender)
+        rollingPolicy.start()
+        rollingFileAppender.rollingPolicy = rollingPolicy
+
+        val triggeringPolicy = SizeBasedTriggeringPolicy<ILoggingEvent>()
+        triggeringPolicy.context = loggerContext
+        triggeringPolicy.maxFileSize = FileSize(maximumFileSize.toLong())
+        triggeringPolicy.start()
+        rollingFileAppender.setTriggeringPolicy(triggeringPolicy)
+      }
+
+      val encoder = PatternLayoutEncoder()
+      encoder.context = loggerContext
+      encoder.charset = StandardCharsets.UTF_8
+      encoder.pattern = "%d{yyyy/MM/dd HH:mm:ss.SSS} %-5level %msg%n"
+      encoder.start()
+
+      rollingFileAppender.encoder = encoder
+      rollingFileAppender.start()
+
+      val root = LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME) as ch.qos.logback.classic.Logger
+      root.level = Level.DEBUG
+      root.detachAndStopAllAppenders()
+      root.addAppender(rollingFileAppender)
+
+      configureOptions = options
+    }
+
+    fun d(tag: String, message: String) {
+      if (configureOptions != null) {
```

I wonder if this has not been configured we should throw an error as I see no point on using this library just to use it as a replacement of `Log`

---

I will add a warning to the logs. I don't want to throw an error just to be conservative on this (in case it is configured in the wrong place, or anything).